students/cjfortu/lesson09/mailroom_00/donor_models.py

```diff
@@ -0,0 +1,93 @@
+#!/usr/bin/env python
+
+"""
+The Mailroom assignment, donor models.
+
+Donor object creation is under the Donor class, while assembling the donor objects is in the DonorCollection class.
+The donor object collection structure is:
+{'name": object'}
+In words, it is a dict with each donor name in string form as the keys, and the donor object as the value.
+
+The remaining features follow the assignment instructions.
+"""
+
+
+class Donor:
+    """Class that holds and accesses information about a single donor."""
+    new_donations = []
+    
+    def __init__(self, donor_name=None, *donations):
+        """Initiate a donor by name and donations."""
+        self.name = donor_name
+        self.donations = [*donations]
+        self.new_donations = [*donations]
+
+    @property
+    def number_donations(self):
+        return len(self.donations)
+    
+    @property
+    def lifetime_donations_sum(self):
+        return sum(self.donations)
+    
+    @property
+    def average_donations(self):
+        return self.lifetime_donations_sum / self.number_donations
+        
+    def add_donations(self, new_donations):
+        """Append new donations to the existing donations attribute."""
+        [self.donations.append(donation) for donation in new_donations]
```

Comment:
this is an invalid use of list comprehension

a list comprehension should produce an output that would get stored in a variable, right now you are generation a list of None values and not store it anywhere.

Reply:
Commenting this line out does cause my tests to fail, including "test_add_donations".  I looked at that test again, and it seems valid.  

Also, I manually run this comprehension in command line, and it achieves the intended effect of looping to append the list represented by **self.donations**.  

Could you help me understand more?  It seems to work, so is it bad Pythonic practice but functional?  Thank you.

Reply:
Correct it is bad practice.

```
>>> l = []
>>> [l.append(i) for i in range(3)]
[None, None, None]
```
 you are allocating an array of 3 elements with None values.

that code should be converted to a regular for loop OR you don't need a loop there at all:
```
self.donations.extend(new_donations)
```